packages/ui/app/pages/Commands.tsx

```diff
@@ -5,16 +5,23 @@ import { notifications } from '@mantine/notifications';
 import { useQuery } from '@tanstack/react-query';
 import React from 'react';
 import { CommandHistoryTable } from '../components/CommandHistoryTable';
+import { useCommandsWs } from '../hooks/useCommandsWs';
 
 export default function Commands() {
   const [command, setCommand] = React.useState('');
   const [target, setTarget] = React.useState('');
   const [activeTab, setActiveTab] = React.useState('history');
 
+  const { connected } = useCommandsWs((msg) => {
+    if (msg.type === 'command_status' || msg.type === 'command_output') {
+      query.refetch();
```

Comment:
this needs to be throttled (or implement some dynamic update instead of doing a full refetch)